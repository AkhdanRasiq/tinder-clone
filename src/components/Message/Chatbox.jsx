import React from 'react'

class Chatbox extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      strTest: ""
    }
  }

  render() {
    return(
      <div>
        
      </div>
    )
  }
}

export default Chatbox
