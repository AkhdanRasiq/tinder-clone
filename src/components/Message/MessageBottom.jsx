import React from 'react'

import TextField from '@mui/material/TextField'
import SendIcon from '@mui/icons-material/Send'
import { IconButton } from "@mui/material"

import { onSendWebsocket } from '../../utils/adapters'
import { handleGenerateMessage } from '../../utils/util'

class MessageBottom extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      strMessage: "",
      arrMessages: []
    }
  }

  componentDidUpdate = (prevProps) => {
    if(this.props.a_arrMessages !== prevProps.a_arrMessages)
      this.setState({ arrMessages: this.props.a_arrMessages })
  }

  handleMessageToSend = () => {
    const generatedMessage = handleGenerateMessage(this.state.strMessage)
    onSendWebsocket(generatedMessage);

    this.setState({ arrMessages: [...this.state.arrMessages, generatedMessage]})
    this.props.a_messageOnUpdateCallback([...this.state.arrMessages, generatedMessage])

    this.setState({ strMessage: "" })
  }

  handleSendMessage = (event) => {
    if (event.key === 'Enter' && this.state.strMessage !== '')
      this.handleMessageToSend()
  }

  handleBtnSendMessage = (event) => {
    event.preventDefault()
    if (this.state.strMessage !== '')
      this.handleMessageToSend()
  }

  handleTyping = (event) => {
    this.setState({ strMessage: event.target.value });
  }

  render() {
    return (
      <div className    ='messageBottomContainer'>

        <TextField
          fullWidth
          className     ='messageBottom_textfield'
          placeholder   ='Type a message'
          autoComplete  ='off'
          onKeyDown     ={this.handleSendMessage} 
          onChange      ={this.handleTyping}
          value         ={this.state.strMessage}  
        />

        <IconButton onClick={this.handleBtnSendMessage}>
          <SendIcon
            className   ='messageBottom_btnSend'
            fontSize    ='large'
          />
        </IconButton>
      </div>
    )
  }
}

export default MessageBottom
