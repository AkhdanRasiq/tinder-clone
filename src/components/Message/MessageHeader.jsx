import React from 'react'

import { Avatar } from '@mui/material'
import { styled } from '@mui/material/styles'
import Badge from '@mui/material/Badge'

const StyledBadgeOnline = styled(Badge)(({ theme }) => ({
  '& .MuiBadge-badge': {
    backgroundColor: '#44b700',
    color: '#44b700',
    width: '10px',
    height: '10px',
    borderRadius: '100%',
    right: 10,
    bottom: 15,
    boxShadow: `0 0 0 2px ${theme.palette.background.paper}`,
    '&::after': {
      position: 'absolute',
      top: -1,
      left: -1,
      width: '100%',
      height: '100%',
      borderRadius: '100%',
      animation: 'ripple 1.2s infinite ease-in-out',
      border: '1px solid currentColor',
      content: '""',
    },
  },
  '@keyframes ripple': {
    '0%': {
      transform: 'scale(.8)',
      opacity: 1,
    },
    '100%': {
      transform: 'scale(2.4)',
      opacity: 0,
    },
  },
}));

const StyledBadgeOffline = styled(Badge)(({ theme }) => ({
  '& .MuiBadge-badge': {
    backgroundColor: '#b70000',
    color: '#b70000',
    width: '10px',
    height: '10px',
    borderRadius: '100%',
    right: 10,
    bottom: 15,
    boxShadow: `0 0 0 2px ${theme.palette.background.paper}`
  }
}));

class MessageHeader extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      isOnline: false
    }
  }

  componentDidMount() {
    window.addEventListener("authorOnline", this.handleEventStatus);
  }

  componentWillUnmount() {
    window.removeEventListener("authorOnline", this.handleEventStatus);
  }

  handleEventStatus = (event) => {
    this.setState({ isOnline: event.detail.isOnline })
  }

  render() {
    return (
      <div className='messageHeaderContainer'>
        {this.state.isOnline ?
          <StyledBadgeOnline
            overlap="circular"
            anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
            variant="dot"
          >
            <Avatar className='messageHeaderAvatar' alt="Avatar Image" src="https://www.logo.wine/a/logo/Tinder_(app)/Tinder_(app)-Flame-Logo.wine.svg" />
          </StyledBadgeOnline>
        :
          <StyledBadgeOffline
            overlap="circular"
            anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
            variant="dot"
          >
            <Avatar className='messageHeaderAvatar' alt="Avatar Image" src="https://www.logo.wine/a/logo/Tinder_(app)/Tinder_(app)-Flame-Logo.wine.svg" />
          </StyledBadgeOffline>
        }
      </div>
    )
  }
}

export default MessageHeader
