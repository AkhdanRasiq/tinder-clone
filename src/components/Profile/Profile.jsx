import React from 'react'

import { Avatar } from '@mui/material'
import LocationOnIcon from '@mui/icons-material/LocationOn';
import EmailIcon from '@mui/icons-material/Email';
import GitHubIcon from '@mui/icons-material/GitHub';
import LinkedInIcon from '@mui/icons-material/LinkedIn';

function Profile() {
  return (
    <div>
      <Avatar className='sidebarProfileAvatar' alt="Avatar Image" src="https://media-exp1.licdn.com/dms/image/C5103AQHB1GRZq-arKg/profile-displayphoto-shrink_400_400/0/1535513202972?e=1648684800&v=beta&t=NFXC30MwzF0Y3zJUgVJk7d3FMAmAHKCdbnESGrOmpYE" />
      <p className='sidebarProfileName'>AKHDAN RASIQ GUMELAR</p>
      <p className='sidebarProfileAbout'>Experienced Software Engineer with a demonstrated history of working in the computer software industry. Skilled in Mobile Application Development, ReactJS, Python, and JavaScript. Strong engineering professional graduated from SMK Telkom Jakarta.</p>
      <div className='sidebarProfilePersonalInfoContainer'>
        <LocationOnIcon fontSize='large' className='sidebarProfilePersonalInfoIcon' />
        <p className='sidebarProfilePersonalInfoText'>Indonesia, Jakarta</p>
      </div>
      <div className='sidebarProfilePersonalInfoContainer'>
        <EmailIcon fontSize='large' className='sidebarProfilePersonalInfoIcon' />
        <p className='sidebarProfilePersonalInfoText'>akhdan.rasiq@gmail.com</p>
      </div>
      <div className='sidebarProfilePersonalInfoContainer'>
        <GitHubIcon fontSize='large' className='sidebarProfilePersonalInfoIcon' />
        <p className='sidebarProfilePersonalInfoText'>github.com/AkhdanRasiq</p>
      </div>
      <div className='sidebarProfilePersonalInfoContainer'>
        <LinkedInIcon fontSize='large' className='sidebarProfilePersonalInfoIcon' />
        <p className='sidebarProfilePersonalInfoText'>linkedin.com/in/AkhdanRasiq</p>
      </div>
    </div>
  )
}

export default Profile
