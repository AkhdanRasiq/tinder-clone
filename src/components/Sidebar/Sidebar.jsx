import React from 'react'
import { Avatar } from '@mui/material'

import { IconButton } from "@mui/material";
import CloseIcon from '@mui/icons-material/Close';
import LocationOnIcon from '@mui/icons-material/LocationOn';
import EmailIcon from '@mui/icons-material/Email';
import GitHubIcon from '@mui/icons-material/GitHub';
import LinkedInIcon from '@mui/icons-material/LinkedIn';

function getWindowDimensions() {
  const { innerWidth: width, innerHeight: height } = window;
  return width
}

class Sidebar extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      bIsActive: false
    }
  }

  componentDidUpdate = (prevProps) => {
    if(this.props.a_bIsActive !== prevProps.a_bIsActive)
        this.setState({ bIsActive: this.props.a_bIsActive })
  }

  handleSidebarCoverHide = (event) => {
    this.setState({ bIsActive: !this.state.bIsActive })
    const eventDispatch = new CustomEvent('sidebarCover', { bIsActive: !this.state.bIsActive });
    window.dispatchEvent(eventDispatch)
  }

  render() {
    const styleSidebarCoverShow = {
      '--display': 'flex'
    }
    const styleSidebarCoverHide = {
      '--display': 'none'
    }
    const styleProfileSidebarHide = getWindowDimensions() > 885 ? {
      '--margin-left': '-30vw'
    } : {
      '--margin-left': '-100vw'
    }
    const styleProfileSidebarShow = {
      '--margin-left': '0vw'
    }

    return (
      <div className='sidebar'>

        <div className='sidebarCover'
          style={this.state.bIsActive ? styleSidebarCoverShow : styleSidebarCoverHide}
          onClick={this.handleSidebarCoverHide}
        />
        <div className='sidebarCard' style={this.state.bIsActive ? styleProfileSidebarShow : styleProfileSidebarHide}>
          <div className='sidebarProfileCloseContainer'>
            <IconButton onClick={this.handleSidebarCoverHide}>
              <CloseIcon fontSize="large" className="headerIcon" />
            </IconButton>
          </div>

          <Avatar className='sidebarProfileAvatar' alt="Avatar Image" src="https://media-exp1.licdn.com/dms/image/C5103AQHB1GRZq-arKg/profile-displayphoto-shrink_400_400/0/1535513202972?e=1648684800&v=beta&t=NFXC30MwzF0Y3zJUgVJk7d3FMAmAHKCdbnESGrOmpYE" />
          <p className='sidebarProfileName'>AKHDAN RASIQ GUMELAR</p>
          <p className='sidebarProfileAbout'>Experienced Software Engineer with a demonstrated history of working in the computer software industry. Skilled in Mobile Application Development, ReactJS, Python, and JavaScript. Strong engineering professional graduated from SMK Telkom Jakarta.</p>
          <div className='sidebarProfilePersonalInfoContainer'>
            <LocationOnIcon fontSize='large' className='sidebarProfilePersonalInfoIcon'/>
            <p className='sidebarProfilePersonalInfoText'>Indonesia, Jakarta</p>
          </div>
          <div className='sidebarProfilePersonalInfoContainer'>
            <EmailIcon fontSize='large' className='sidebarProfilePersonalInfoIcon'/>
            <p className='sidebarProfilePersonalInfoText'>indan.gumelar20@gmail.com</p>
          </div>
          <div className='sidebarProfilePersonalInfoContainer'>
            <GitHubIcon fontSize='large' className='sidebarProfilePersonalInfoIcon'/>
            <p className='sidebarProfilePersonalInfoText'>github.com/AkhdanRasiq</p>
          </div>
          <div className='sidebarProfilePersonalInfoContainer'>
            <LinkedInIcon fontSize='large' className='sidebarProfilePersonalInfoIcon'/>
            <p className='sidebarProfilePersonalInfoText'>linkedin.com/in/AkhdanRasiq</p>
          </div>
        </div>

        {/* <div className='sidebarCover'
          style={styleSidebarCoverShow}
        />
        <div className='sidebarCard' style={styleProfileSidebarShow}>
          <Avatar className='sidebarProfileAvatar' alt="Remy Sharp" src="https://media-exp1.licdn.com/dms/image/C5103AQHB1GRZq-arKg/profile-displayphoto-shrink_400_400/0/1535513202972?e=1648684800&v=beta&t=NFXC30MwzF0Y3zJUgVJk7d3FMAmAHKCdbnESGrOmpYE" />
          <p className='sidebarProfileName'>AKHDAN RASIQ GUMELAR</p>
          <p className='sidebarProfileAbout'>Experienced Software Engineer with a demonstrated history of working in the computer software industry. Skilled in Mobile Application Development, ReactJS, Python, and JavaScript. Strong engineering professional graduated from SMK Telkom Jakarta.</p>
          <div className='sidebarProfilePersonalInfoContainer'>
            <LocationOnIcon fontSize='large'/>
            <p className='sidebarProfilePersonalInfoText'>Indonesia, Jakarta</p>
          </div>
          <div className='sidebarProfilePersonalInfoContainer'>
            <EmailIcon fontSize='large'/>
            <p className='sidebarProfilePersonalInfoText'>indan.gumelar20@gmail.com</p>
          </div>
          <div className='sidebarProfilePersonalInfoContainer'>
            <GitHubIcon fontSize='large'/>
            <p className='sidebarProfilePersonalInfoText'>github.com/AkhdanRasiq</p>
          </div>
          <div className='sidebarProfilePersonalInfoContainer'>
            <LinkedInIcon fontSize='large'/>
            <p className='sidebarProfilePersonalInfoText'>linkedin.com/in/AkhdanRasiq</p>
          </div>
        </div> */}

      </div>
    )
  }
}

export default Sidebar
