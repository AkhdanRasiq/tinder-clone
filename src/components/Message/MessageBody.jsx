import React from 'react'
import Chatbox from './Chatbox'

import { goToBottom } from '../../utils/util'
import { currentWebsocket } from '../../utils/adapters'

class MessageBody extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      arrMessages: []
    }
  }

  componentDidMount = () => {
    this.handleResize()
    window.addEventListener('resize', this.handleResize)
  }

  componentWillUnmount = () => {
    window.removeEventListener('resize', this.handleResize)
  }

  componentDidUpdate = (prevProps) => {
    if(this.props.a_arrMessages !== prevProps.a_arrMessages) {
      this.setState({ arrMessages: this.props.a_arrMessages })
    }
    if(currentWebsocket().readyState === WebSocket.OPEN)
      goToBottom()
  }

  handleResize = () => {
    let vh = window.innerHeight * 0.01
    document.documentElement.style.setProperty('--vh', `${vh}px`)
    goToBottom()
  }

  render() {
    return (
      <div className='messageBodyContainer' id='messageBodyContainer'>
        {this.state.arrMessages.map((msg) => (
          <Chatbox
            key={msg.timestamp}
            a_messages={msg}
          />
        ))}
      </div>
    )
  }
}

export default MessageBody
