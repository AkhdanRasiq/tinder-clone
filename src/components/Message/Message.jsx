import React, { Component } from 'react'
import { connectWebsocket } from '../../utils/adapters'

import MessageHeader from '../Message/MessageHeader'
import MessageBody from '../Message/MessageBody'
import MessageBottom from '../Message/MessageBottom'

class Message extends Component {
  constructor(props) {
    super(props)
    this.state = {
      arrMessages: []
    }
  }

  componentDidMount = () => {
    connectWebsocket()
    window.addEventListener("newMessage", this.handleEventNewMessage);
  }

  componentWillUnmount() {
    window.removeEventListener("newMessage", this.handleEventNewMessage);
  }

  handleEventNewMessage = (event) => {
    this.setState({ arrMessages: [...this.state.arrMessages, event.detail] })
  }

  handleMessagesUpdate = (event) => {
    this.setState({ arrMessages: event })
  }

  render() {
    return (
      <div className="messageContainer">
        <MessageHeader />
        <MessageBody
          a_arrMessages             = {this.state.arrMessages}
        />

        <MessageBottom
          a_arrMessages             = {this.state.arrMessages}
          a_messageOnUpdateCallback = {this.handleMessagesUpdate}
        />
      </div>
    )
  }
}

export default Message
