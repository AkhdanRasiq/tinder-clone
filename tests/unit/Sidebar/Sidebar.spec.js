import React from 'react'
import { shallow } from "enzyme"

import Sidebar from "../../../src/components/Sidebar/Sidebar"
import Profile from "../../../src/components/Profile/Profile"
import Message from "../../../src/components/Message/Message"

describe("Sidebar.jsx", () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(
      <Sidebar
        a_bPersonIsActive={false}
        a_bMessageIsActive={false}
      />
    )
  })

  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot()
  })
  
  it("Profile & Message renders properly", () => {
    expect(wrapper.contains(<Profile />)).toBe(true)
    expect(wrapper.contains(<Message />)).toBe(true)
  })
})
