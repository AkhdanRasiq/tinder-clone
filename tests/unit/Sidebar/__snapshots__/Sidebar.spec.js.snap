// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar.jsx renders properly 1`] = `ShallowWrapper {}`;
